### PR TITLE
feat(stepper)!: stepper migrate tokens

### PR DESCRIPTION
**Pull request by mlogsdon18**
## Description

This utilizes the In-field button tokens and Figma file, along with the Textfield tokens, to migrate the Stepper component. This PR relies on the In-field button PR, so that will need to be merged first.

## To-do list

- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.
- [x] I have updated any relevant storybook stories and templates.
- [x] If my change(s) include visual change(s), a designer has reviewed and approved those changes.

- [x] In-field tokens have been added to the main tokens release
- [ ] This pull request is ready to merge.

### CODE REVIEW

**Review by jenndiaz — CHANGES_REQUESTED** [review]
Nice job, theres a lot of details in each of these. I had a couple of small questions. 

Do we also want to handle the button text in this component with an aria label on the button, like you did in in-field button ?

**Comment by jenndiaz** [comment]
@mlogsdon18 It looks like something regressed with the keyboard focus indicator. 

It is not displaying on Storybook or when tabbing though the docs site.

File: components/stepper/stories/template.js [hunk 1]
```diff
@@ -50,6 +67,7 @@ export const Template = ({
 		>
 			${Textfield({
 				...globals,
+				size,
 				placeholder: "1",
```
Comment (jawinn) [inline components/stepper/stories/template.js:71]: Should we remove placeholder from this example? We don't have one in the docs and it makes it look like the field value is 1. Maybe set a default value of 1 or 0 in the field instead?

I also noticed that the input being rendered in Storybook isn't `type="number"` (allows letters to be inputted).

**Review by jawinn — CHANGES_REQUESTED** [review]
I tried putting in a longer value and it can get obscured by going underneath the arrow buttons. The right padding probably needs an adjustment to add the width of the arrows. The prod version doesn't appear to do this currently.
Screenshot 2023-09-11 at 2 15 38 PM

File: components/stepper/stories/template.js (hunk 1 above)
Comment (mlogsdon18) [reply components/stepper/stories/template.js:70]: Good catches there! I removed placeholder and am passing the number type now to textfield.

File: components/stepper/themes/express.css [hunk 2]
```diff
@@ -11,14 +11,23 @@ governing permissions and limitations under the License.
...
-  /* @todo remove this when stepper is migrated; this ensures stepper buttons inherit border color and width */
   .spectrum-Stepper {
-    --spectrum-stepper-border-width: var(--spectrum-border-width-200); /* express stepper has border */
+    --spectrum-stepper-border-width: var(--spectrum-border-width-200);
+
+    --spectrum-stepper-buttons-border-style: solid;
+    --spectrum-stepper-buttons-border-width: var(--spectrum-border-width-200);
+    --spectrum-stepper-buttons-background-color: var(--spectrum-gray-50);
+
+    --spectrum-stepper-button-border-width: 0;
```
Comment (pfulton) [inline components/stepper/themes/express.css:21]: On this line, `-button-` is singular in the name, but in `themes/spectrum.css` it looks like it's plural `-buttons-` and I think `-buttons` is what we refer to in the `index.css`.

Can you take a look to make sure that the naming is correct?

**Review by pfulton — CHANGES_REQUESTED** [review]
Just one thing I noticed here, @mlogsdon18

File: components/stepper/themes/express.css (hunk 2 above)
Comment (mlogsdon18) [reply components/stepper/themes/express.css]: You're right! It doesn't look like that variable is being used anymore (along with a few others) now that I'm using in-field button so I've removed them

**Comment by pfulton** [comment]
Beta published: `@spectrum-css/stepper@5.0.0-beta.0`

**Review by pfulton — APPROVED** [review]

**Review by jenndiaz — APPROVED** [review]

**Review by jawinn — APPROVED** [review]

**Pull request merged**